cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -468,7 +468,7 @@ private Metadata getSnapshotBeforeTime(NamespacedEntityId targetId, long timeMil
       return Collections.emptySet();
     }
 
-    List<ImmutablePair<byte [], byte []>> fuzzyKeys = new ArrayList<>();
+    List<ImmutablePair<byte [], byte []>> fuzzyKeys = new LinkedList<>();
```

Comment:
Why the change? Also, since we know the expected size (`targetIds.size()`), we should use `new ArrayList<>(targetIds.size())`.